app/src/main/java/io/pslab/fragment/GasSensorDataFragment.java

```diff
@@ -397,6 +397,7 @@ private void visualizeData() {
                 mChart.invalidate();
             }
         }
+        writeLogToFile(System.currentTimeMillis(), (float) ppmValue);
```

Comment:
Do we need to cast the double to float? What if the variable is set to float in the first place?

Reply:
https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.3